### PR TITLE
[processor/metricstransformprocessor]: Support median aggregation type

### PR DESCRIPTION
**Pull request by odubajDT**
**Description:** <Describe what has changed.>
- use aggregation business logic from `interval/core`
- support `median` aggregation type
- testing
- docs

**Link to tracking Issue:** #16224 

**Depends on** https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33669

### CODE REVIEW

File: processor/metricstransformprocessor/metrics_transform_processor_otlp.go [hunk 1]
```diff
@@ -541,7 +554,13 @@ func transformMetric(metric pmetric.Metric, transform internalTransform) bool {
...
 		case aggregateLabels:
 			if canChangeMetric {
-				aggregateLabelsOp(metric, op)
+				ops := []string{}
+				for k, v := range op.labelSetMap {
+					if v {
+						ops = append(ops, k)
+					}
+				}
+				aggregateLabelsOp(metric, ops, op.configOperation.AggregationType)
```
Comment (evan-bradley) [inline processor/metricstransformprocessor/metrics_transform_processor_otlp.go:563]:
```suggestion
				attrs := []string{}
				for k, v := range op.labelSetMap {
					if v {
						attrs = append(attrs, k)
					}
				}
				aggregateLabelsOp(metric, attrs, op.configOperation.AggregationType)
```
This is a list of attributes and not operations, right?

**Review by evan-bradley — APPROVED** [review]
Looks good to me based on the fact that the tests have been minimally changed.

**Comment by evan-bradley** [comment]
@dmitryax Could you take a look? This bridges what I believe is the final gap between the metrics transform processor and transform processor, and will let us confidently merge https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33334.

I'll do a review after this to confirm, but I think once we know there is parity between the two, we can start pointing users to the transform processor as a meaningful replacement.

**Review by dmitryax — APPROVED** [review]
LGTM

**Pull request merged**